crates/rolldown/tests/fixtures/tree_shaking/multi-declarator/foo.js

```diff
@@ -1,2 +1,2 @@
-export const foo = 'foo', bar = 'bar'
+export const { baz } = { baz: 'baz' }, foo = 'foo', bar = 'bar';
```

Comment:
`{baz}` is shaked, as expected.

Reply:
You probably need to consider the potential side-effect when deconstructing from object or array.

Reply:
Yeah, looks like esbuild only marks **object destructing** as side effects. https://esbuild.github.io/try/#YgAwLjIwLjIALS1idW5kbGUgLS1taW5pZnk9ZmFsc2UgLS1mb3JtYXQ9ZXNtAGUAZW50cnkuanMAaW1wb3J0IHsgZm9vLCBjfSBmcm9tICcuL2ZvbycKCgphc3NlcnQuZXF1YWwoZm9vLCAnZm9vJykKAABmb28uanMAZXhwb3J0IGNvbnN0IHtiYXp9ID0ge2JhejogJyd9LCBmb28gPSAnZm9vJywgYmFyID0gJ2JhcicKZXhwb3J0IGNvbnN0IFtjLCBkXSA9IFsxLCAyXQ

Reply:
Nah. It's tricky: https://esbuild.github.io/try/#YgAwLjIwLjIALS1idW5kbGUgLS1taW5pZnk9ZmFsc2UgLS1mb3JtYXQ9ZXNtAGUAZW50cnkuanMAaW1wb3J0IHsgZm9vLCBjfSBmcm9tICcuL2ZvbycKCgphc3NlcnQuZXF1YWwoZm9vLCAnZm9vJywpCgAAZm9vLmpzAGNvbnN0IGRlbW8gPSBbMSwgMl0KZXhwb3J0IGNvbnN0IHtiYXp9ID0ge2JhejogJyd9LCBmb28gPSAnZm9vJywgYmFyID0gJ2JhcicKZXhwb3J0IGNvbnN0IFtjLCBkXSA9IGRlbW8

Reply:
Yeah, I need to read spec carefully